src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,401 @@
+    private static final String MANUAL_DIR = FilenameUtils
+        .separatorsToSystem("manual/");
+    private static final String PACKAGE_TEST_NAME = "test";
+    private static final int MAX_DEPTH_FOR_DIRS = 1;
+
+    private TestClassesDTO currentClassesDTO;
+    private List<TestClassesDTO> caseClassesList;
+
+    @SuppressWarnings("Duplicates")
+    private PageObjectsGenerator initPog(String jsonPath, String url,
+                                         boolean checkLocatorUniqueness,
+                                         boolean forceGenerateFiles) throws IOException {
+        List<String> urls = new ArrayList<>();
+        urls.add(url);
+        SupportedTypesContainer bc = new SupportedTypesContainer();
+        JsonRuleMapper parser = new JsonRuleMapper(new File(jsonPath), new ObjectMapper());
+        XpathToCssTransformation xpathToCssTransformation = new XpathToCssTransformation();
+        JavaPoetAdapter javaPoetAdapter = new JavaPoetAdapter(bc, xpathToCssTransformation);
+
+        ValidatorsStarter validatorsStarter = new ValidatorsStarter(bc);
+        validatorsStarter.setCheckLocatorsUniqueness(checkLocatorUniqueness);
+
+        PageObjectsGenerator pog = new PageObjectsGenerator(parser, validatorsStarter,
+            javaPoetAdapter, RESOURCE_DIR, urls, PACKAGE_TEST_NAME);
+        pog.setForceGenerateFile(forceGenerateFiles);
+        return pog;
+    }
+
+    public MainIntegrationTest(List<TestClassesDTO> list) {
+        this.caseClassesList = list;
+    }
+
+    /**
+     * Forms List<Object[]> which contains parameters for testing. Every list's element
+     * (it's Object array) contains one parameter - list of TestClassesDTO objects. This list
+     * represents each test case. For this purpose data() method scans
+     * through scr/test/resources/manual to find all test cases dirs and then form list
+     * of TestClassesDTO objects for every test case
+     *
+     * About regex: .+\.[a-z]+$
+     * This regex is used for matching files with extensions (in this particular situation for
+     * matching .java or .properties files). Example:
+     * src/test/resources/manual/google/Google.java - matches
+     * src/test/resources/manual/google             - doesn't match
+     *
+     * @return list of parameters set for every test case
+     */
+    @Parameters
+    public static Iterable<Object[]> data() throws IOException, ClassNotFoundException {
+
+        List<Path> caseDirs = getCaseDirsNames(
+            RESOURCE_DIR + MANUAL_DIR,
+            MAX_DEPTH_FOR_DIRS
+        );
+        List<Path> insideCaseDirs;
+        List<Path> javaFilesList;
+        List<Path> casePropertiesList;
+        List<Object[]> testParameters = new ArrayList<>();
+        Properties caseProperties;
+        List<TestClassesDTO> objectList;
+
+        for (Path caseDir : caseDirs) {
+            objectList = new ArrayList<>();
+            caseProperties = new Properties();
+
+            insideCaseDirs = Files.find(
```

Comment:
By trying to reusing one regex for each case you are actually making code harder to read.
This code fragment can be rewritten using following code:

    Files.list(caseDir)
                .filter(s -> Files.isDirectory(s))
                .collect(Collectors.toList());